test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -398,6 +405,20 @@ object Assertion extends AssertionVariants {
       }
     }
 
+  /**
+   * Makes a new assertion that requires an Iterable contain at least one of the
+   * specified elements.
+   */
+  def hasAtLeastOneOf[A](values: Set[A]): Assertion[Iterable[A]] =
```

Comment:
Since `intersection` is defined for types other than `Set` should we change the `values` type to `Iterable` here?

Reply:
All assertions based on the intersection of two iterables have now been changed to accept an `Iterable`.